yt/visualization/fits_image.py

```diff
@@ -84,6 +104,27 @@ def __init__(self, data, fields=None, units=None, width=None, wcs=None):
         exclude_fields = ['x','y','z','px','py','pz',
                           'pdx','pdy','pdz','weight_field']
 
+        if isinstance(data, _astropy.pyfits.HDUList):
+            self.hdulist = data
+            for hdu in data:
+                self.fields.append(hdu.header["btype"])
+                self.field_units[hdu.header["btype"]] = hdu.header['bunit']
+
+            self.shape = self.hdulist[0].shape
+            self.dimensionality = len(self.shape)
+            wcs_names = [key for key in self.hdulist[0].header 
+                         if "WCSNAME" in key]
+            for name in wcs_names:
```

Comment:
For my own edification, is the `[-1]` here because it's `WCSName` and then a single character?